docs/planning/otel/OTelWebSdk.md

```diff
@@ -0,0 +1,1567 @@
+sdk.addLogRecordProcessor(new SimpleLogRecordProcessor(exporter));
+
+// 2. Lightweight telemetry initializers (optimized for performance-critical scenarios)
+sdk.addTelemetryInitializer((item) => {
+  item.data.customProperties['app.version'] = '2.1.0';
+  return true;
+});
+```
+
+The telemetry processing pipeline executes **OpenTelemetry processors first**, followed by lightweight telemetry initializers, ensuring standards compliance while providing performance-optimized processing for common scenarios.
+
+## Multi-Instance SDK Factory Implementation
+
+### Core Factory Interface
+
+The multi-instance SDK factory is implemented through a sophisticated interface that coordinates multiple SDK instances while optimizing resource usage.
+
+For complete interface definitions, see [OTelWebSdk-Interfaces.md](./specs/OTelWebSdk-Interfaces.md).
+
+
+### Factory Access Patterns
+
+The SDK provides multiple factory access patterns to support different deployment scenarios:
+
+- **Synchronous Factory Access**: For immediate SDK access
+- **Asynchronous Factory Access**: For CDN loading and version fetching
+- **Named Factory Pattern**: For multi-project scenarios
+- **Direct Factory Creation**: For custom scenarios
+
+```
+
+### Version-Aware and Async Factory Loading
+
+The factory supports both synchronous and asynchronous loading patterns with version specification and compatibility checking.
+
+#### **Synchronous Factory Access**
+
+```typescript
+// Basic usage - gets existing factory or creates new one
+const factory = createSdkFactory();
+console.log(`Using SDK version: ${factory.version}`);
+
+// Version-aware access with options
+const factory = createSdkFactory({
+  version: '1.5.0',                    // Request specific version
+  versionStrategy: 'compatible',       // Accept compatible versions
+  allowFallback: true                  // Fall back if exact version unavailable
+});
+
+// Verify version compatibility
+if (factory.version !== '1.5.0') {
+  console.warn(`Using ${factory.version} instead of requested 1.5.0`);
+}
+```
+
+#### **Asynchronous Factory Access**
+
+```typescript
+// Async loading for CDN scenarios
+const factory = await createSdkFactoryAsync();
+
+// Request specific version with CDN loading
+const factory = await createSdkFactoryAsync({
+  version: '1.5.0',
+  versionStrategy: 'exact',
+  timeout: 10000,                      // 10 second timeout
+  cdnBaseUrl: 'https://custom-cdn.com/otel-web-sdk'
+});
+
+// Handle loading failures
+try {
+  const factory = await createSdkFactoryAsync({
+    version: '2.0.0',
+    versionStrategy: 'exact',
+    allowFallback: false
+  });
+} catch (error) {
+  console.error('Failed to load exact version 2.0.0:', error.message);
+  // Fallback to any available version
+  const factory = await createSdkFactoryAsync({ versionStrategy: 'any' });
+}
+```
+
+#### **Version Strategy Behaviors**
+
+```typescript
+interface IVersionMatchResult {
```

Comment:
I don't think we want to try and implement this version loading stuff as part of v1, maybe v5